ACPs/77-reinventing-subnets/README.md

```diff
@@ -38,31 +38,39 @@ With no Elastic Subnets live on Mainnet, it is clear that Permissionless Subnets
-+-----------+----------+-----------+
-|    weight :   uint64 |   8 bytes |
-+-----------+----------+-----------+
-| timestamp :   uint64 |   8 bytes |
-+-----------+----------+-----------+
-| signature : [64]byte |  64 bytes |
-+-----------+----------+-----------+
-                       | 144 bytes |
-                       +-----------+
++------------+----------+-----------+
+|   subnetID : [32]byte |  32 bytes |
++------------+----------+-----------+
+|     nodeID : [32]byte |  32 bytes |
++------------+----------+-----------+
+|     weight :   uint64 |   8 bytes |
++------------+----------+-----------+
+| expiryTime :   uint64 |   8 bytes |
++------------+----------+-----------+
+|  signature : [64]byte |  64 bytes |
++------------+----------+-----------+
+                        | 144 bytes |
+                        +-----------+
 ```
 
-- `nodeID`, `subnetID` and `weight` are for the Subnet Validator being added
-- `timestamp` is the time after which this message is invalid. After `timestamp` elapses on the P-Chain, this Avalanche Warp Message can no longer be used to add the `nodeID` to the validator set of `subnetID`. For replay protection, the P-Chain will store used `messageID`s (hash of the entire Avalanche Warp Message). To prevent the P-Chain from having to store an unbounded number of `messageID`s, the `timestamp` is required to be no longer than 48 hours in the future of the time the transaction is issued on the P-Chain.
-- `signature` is the raw bytes of the Ed25519 signature over the concatenated bytes of `[subnetID]+[nodeID]+[blsPublicKey]+[weight]+[timestamp]`. This signature must correspond to the Ed25519 public key that is used for the `nodeID`. This approach prevents NodeIDs from being unwillingly added to Subnets.
+- `subnetID`, `nodeID`, and `weight` are for the Subnet Validator being added
+- `expiryTime` is the time after which this message is invalid. After the P-Chain timestamp is past `expiryTime`, this Avalanche Warp Message can no longer be used to add the `nodeID` to the validator set of `subnetID`. For replay protection, the P-Chain will store used `messageID`s (hash of the entire Avalanche Warp Message). To prevent the P-Chain from having to store an unbounded number of `messageID`s, the `expiryTime` is required to be no longer than 48 hours in the future of the time the transaction is issued on the P-Chain.
+- `signature` is the raw bytes of the Ed25519 signature over the concatenated bytes of `[subnetID]+[nodeID]+[blsPublicKey]+[weight]+[balance]+[expiryTime]`. This signature must correspond to the Ed25519 public key that is used for the `nodeID`. This approach prevents NodeIDs from being unwillingly added to Subnets. `balance` is the minimum initial $AVAX balance that must be attached to the validator serialized as a uint64. The `signature` field will be validated by the P-Chain in Step 2. A Subnet may choose to validate that the `signature` field is well-formed but it is not required.
```

Comment:
What is the issue with a NodeID being unwillingly added to a Subnet again? If they aren't participating/or haven't staked, what is the issue?

Reply:
It's the other side that's the concern: If a NodeID has locked up stake on the Subnet and wants to be a Subnet Validator, somebody may take the BLS signature and instantiate the validator on the P-Chain with a small balance to kick them out immediately. With the Ed25519 signature attestation of a `balance`, the validator knows that they must be instantiated on the P-Chain with a `Balance` >= `balance`.